homeassistant/components/surepetcare/__init__.py

```diff
@@ -165,3 +114,60 @@ async def set_lock_state(self, flap_id: int, state: str) -> None:
             await self.surepy.sac.lock_out(flap_id)
         elif state == LockState.LOCKED_ALL.name.lower():
             await self.surepy.sac.lock(flap_id)
+
+    async def async_setup(self) -> bool:
```

Comment:
I feel like stuff this into this class, doesn't make it more readable. I think we should keep this in the setup entry for now, which also reduces the number of unrelated changes in this PR a lot.